pipelines/batch/src/main/java/org/openmrs/analytics/ConvertResourceFn.java

```diff
@@ -15,97 +15,55 @@
 package org.openmrs.analytics;
 
 import java.io.IOException;
+import java.sql.SQLException;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
 import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 
-import ca.uhn.fhir.context.FhirContext;
-import ca.uhn.fhir.parser.IParser;
-import com.google.common.annotations.VisibleForTesting;
 import org.apache.beam.sdk.metrics.Counter;
 import org.apache.beam.sdk.metrics.Metrics;
-import org.apache.beam.sdk.transforms.DoFn;
 import org.hl7.fhir.r4.model.Meta;
 import org.hl7.fhir.r4.model.Resource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
-// TODO: A good amount of functionality/setup here is shared with FetchSearchPageFn. 
-// There is room for refactoring in the future.
-public class ConvertResourceFn extends DoFn<HapiRowDescriptor, Integer> {
+public class ConvertResourceFn extends FetchSearchPageFn<HapiRowDescriptor> {
 	
-	private final HashMap<String, Counter> numFetchedResources;
+	private static final Logger log = LoggerFactory.getLogger(ConvertResourceFn.class);
 	
-	private final HashMap<String, Counter> totalParseTimeMillis;
+	private final SimpleDateFormat simpleDateFormat;
 	
-	private final HashMap<String, Counter> totalGenerateTimeMillis;
+	private final HashMap<String, Counter> numFetchedResourcesMap;
 	
-	private final HashMap<String, Counter> totalPushTimeMillis;
+	private final HashMap<String, Counter> totalParseTimeMillisMap;
 	
-	private final String sinkPath;
+	private final HashMap<String, Counter> totalGenerateTimeMillisMap;
 	
-	private final String sinkUsername;
+	private final HashMap<String, Counter> totalPushTimeMillisMap;
 	
-	private final String sinkPassword;
-	
-	private final String parquetFile;
-	
-	private final int secondsToFlush;
-	
-	private final int rowGroupSize;
-	
-	@VisibleForTesting
-	protected ParquetUtil parquetUtil;
-	
-	protected FhirStoreUtil fhirStoreUtil;
-	
-	private SimpleDateFormat simpleDateFormat;
-	
-	private IParser parser;
-	
-	ConvertResourceFn(FhirEtlOptions options) {
-		this.sinkPath = options.getFhirSinkPath();
-		this.sinkUsername = options.getSinkUserName();
-		this.sinkPassword = options.getSinkPassword();
-		this.parquetFile = options.getOutputParquetPath();
-		this.secondsToFlush = options.getSecondsToFlushParquetFiles();
-		this.rowGroupSize = options.getRowGroupSizeForParquetFiles();
-		
-		this.numFetchedResources = new HashMap<String, Counter>();
-		this.totalParseTimeMillis = new HashMap<String, Counter>();
-		this.totalGenerateTimeMillis = new HashMap<String, Counter>();
-		this.totalPushTimeMillis = new HashMap<String, Counter>();
+	ConvertResourceFn(FhirEtlOptions options, String stageIdentifier) {
```

Comment:
Is it OK to expose the constructor to the public or is it overkill to have a builder method?